app/models/factory.rb

```diff
@@ -80,4 +80,31 @@ def depth
       asso.factory.depth
     end.max + 1
   end
+
+  # Return factory's familty array. index equal to depth.
+  def family
+    queue = []
+    queue.unshift(self)
+    results = []
+    generation = 0
+    while (!queue.empty?)
```

Comment:
Style/ParenthesesAroundCondition: Don't use parentheses around the condition of a `while`.